modules/claims_api/lib/claims_api/v2/disability_compensation_pdf_mapper.rb

```diff
@@ -175,11 +175,109 @@ def get_treatments
           tx['dateOfTreatment'] = tx['startDate']
           tx['doNotHaveDate'] = tx['startDate'].nil?
           tx.delete('center')
-          tx.delete('treatedDisabilityName')
+          tx.delete('treatedDisabilityNames')
           tx.delete('startDate')
           tx
         end
       end
+
+      def service_info
+        symbolize_service_info
+        most_recent_service_period
+        array_of_remaining_service_date_objects
+        confinements
+        national_guard
+        service_info_other_names
+        fed_activation
+
+        @pdf_data
+      end
+
+      def symbolize_service_info
+        @pdf_data[:data][:attributes][:serviceInformation].merge!(
+          @auto_claim['serviceInformation'].deep_symbolize_keys
+        )
+
+        @pdf_data
+      end
+
+      def most_recent_service_period
+        @pdf_data[:data][:attributes][:serviceInformation][:mostRecentActiveService] = {}
+        most_recent_period = @pdf_data[:data][:attributes][:serviceInformation][:servicePeriods].max_by do |sp|
+          sp[:activeDutyEndDate]
+        end
+
+        @pdf_data[:data][:attributes][:serviceInformation][:mostRecentActiveService][:startDate] =
+          most_recent_period[:activeDutyBeginDate]
+        @pdf_data[:data][:attributes][:serviceInformation][:mostRecentActiveService][:endDate] =
+          most_recent_period[:activeDutyEndDate]
+        @pdf_data[:data][:attributes][:serviceInformation][:placeOfLastOrAnticipatedSeparation] =
+          most_recent_period[:separationLocationCode]
+        @pdf_data[:data][:attributes][:serviceInformation][:branchOfService] = most_recent_period[:serviceBranch]
+        @pdf_data[:data][:attributes][:serviceInformation][:serviceComponent] = most_recent_period[:serviceComponent]
+
+        @pdf_data
+      end
+
+      def array_of_remaining_service_date_objects
+        arr = []
+        @pdf_data[:data][:attributes][:serviceInformation][:servicePeriods].each do |sp|
+          arr.push({ startDate: sp[:activeDutyBeginDate], endDate: sp[:activeDutyEndDate] })
+        end
+        sorted = arr.sort_by { |sp| sp[:activeDutyEndDate] }
+        sorted.pop if sorted.count > 1
+        @pdf_data[:data][:attributes][:serviceInformation][:additionalPeriodsOfService] = sorted
+        @pdf_data[:data][:attributes][:serviceInformation].delete(:servicePeriods)
+        @pdf_data
+      end
+
+      def confinements
+        si = {}
+        @pdf_data[:data][:attributes][:serviceInformation][:confinements].map do |confinement|
+          start = confinement[:approximateBeginDate]
+          end_date = confinement[:approximateEndDate]
+          si[:prisonerOfWarConfinement] = { confinementDates: {} }
+          si[:prisonerOfWarConfinement][:confinementDates][:startDate] = start
+          si[:prisonerOfWarConfinement][:confinementDates][:endDate] = end_date
+          si[:confinedAsPrisonerOfWar] = true if start
+          si
+        end
```

Comment:
I'm not totally sure this would handle multiple confinements.  When I run something through it like 
`"confinements": [
       {
           "approximateBeginDate": "2021-11-06",
           "approximateEndDate": "2023-12-09"
        },
                    {
                        "approximateBeginDate": "2016-11-06",
                        "approximateEndDate": "2017-12-09"
                    }
                ],`

the last one just overrides the first.  Obviously totally possible I misunderstand what the output needs to be exactly so just let me know if that is the case but I thought it was worth at least asking about.

Reply:
@rockwellwindsor-va good catch, thanks! Pushing now.